src/sources/aws_kinesis_firehose/errors.rs

```diff
@@ -0,0 +1,87 @@
+use crate::event::Event;
+use snafu::Snafu;
+use warp::http::StatusCode;
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility = "pub")]
+pub enum RequestError {
+    #[snafu(display("X-Amz-Firehose-Access-Key required for request: {}", request_id))]
```

Comment:
This particular error indicates that the incoming HTTP request did not have the access key (passed as `X-Amz-Firehose-Access-Key`) rather than that `vector` didn't have the access key configured. I can see what you are saying about it being an implementation detail though. I'll update the language (though I may still mention `X-Amz-Firehose-Access-Key` for end-user debugging purposes).